### PR TITLE
Fix: KtComments UI Issues 

### PR DESCRIPTION
**Pull request by carsoli**
# Issues
- `KtComment` avatar gets squashed with long comments and comments grow with the content  
Screenshot from 2022-05-24 13-42-27
- `KtComment` "options" (edit - delete) don't skip stacking context
( TODO: include screenshot)
- `KtComment` edit/delete buttons are not translatable
- For Replies, the reply button only shows on hover, and only shows the Icon
  - this is undesired, and a unification of what the "reply" buttons look like for both comments and replies was requested
- By default, the postEscape parser prop is a no-op function 
  - the default desired behavior is to convert new lines to `<br/>`
Screenshot from 2022-05-24 13-47-11
Screenshot from 2022-05-24 13-47-25
- the inline edit field shown when you "edit" a comment can be resized vertically / horizontally
Screenshot from 2022-05-24 13-48-50

# Internal Issues
- code duplication
- styles defined in their own style sheet & unscoped

# Description
- This PR aims to resolve all of the above issues. It should be reviewable by commit, but the commits should be squashed at merge.
- The code duplication has been unified (sub-components were defined) and used on `KtComment` and `CommentReply` level.
- The UI for `Reply` button was unified.
- The translation context was extended to translated edit and delete by default.
- The payload API for edit/delete events was modified such that the `edit` event handles both editing of replies and editing of comments, and the delete event was already implemented as such.
- The comment options use tippy (which skips stacking context) 
- The avatar no longer squishes and the comment no longer grows with content (word-break is preferred)

### CODE REVIEW

**Review by HansAuger — APPROVED** [review]

File: packages/kotti-ui/source/kotti-comment/KtComment.vue [hunk 1]
```diff
@@ -165,16 +132,34 @@ export default defineComponent<KottiComment.PropsInternal>({
...
+		display: flex;
+		flex: 1;
+		flex-direction: column;
+		margin-left: var(--unit-2);
+
+		&__info {
+			display: flex;
+			width: 100%;
+			font-size: 0.7rem;
+			line-height: 1.2rem;
```
Comment (santiagoballadares) [inline packages/kotti-ui/source/kotti-comment/KtComment.vue:161]: Shouldn't this use unit variables?
`font-size: calc(var(--unit-3) + var(--unit-h));`
`line-height: var(--unit-6);`

File: packages/kotti-ui/source/kotti-comment/components/CommentReply.vue [hunk 2]
```diff
@@ -1,131 +1,103 @@
...
+		const isEditing = ref<boolean>(false)
+
 		return {
-			inlineMessageValue: '',
-			isInlineEdit: false,
+			actionOptions: computed(() => {
+				const options = []
+				if (props.isEditable)
+					options.push({
+						label: 'Edit',
```
Comment (santiagoballadares) [inline packages/kotti-ui/source/kotti-comment/components/CommentReply.vue:65]: Shouldn't this use translation strings?
`label: translations.value.editButton`,

File: packages/kotti-ui/source/kotti-comment/components/CommentReply.vue [hunk 3]
```diff
@@ -1,131 +1,103 @@
...
+						onClick: () => {
+							isEditing.value = true
+						},
+					})
+				if (props.isDeletable)
+					options.push({
+						label: 'Delete',
```
Comment (santiagoballadares) [inline packages/kotti-ui/source/kotti-comment/components/CommentReply.vue:72]: Shouldn't this use translation strings?
`label: translations.value.deleteButton,`

File: packages/kotti-ui/source/kotti-comment/components/CommentReply.vue [hunk 4]
```diff
@@ -1,131 +1,103 @@
...
-			inlineMessageValue: '',
-			isInlineEdit: false,
+			actionOptions: computed(() => {
+				const options = []
+				if (props.isEditable)
+					options.push({
+						label: 'Edit',
+						onClick: () => {
+							isEditing.value = true
+						},
```
Comment (santiagoballadares) [inline packages/kotti-ui/source/kotti-comment/components/CommentReply.vue:68]:
```suggestion
						onClick: () => (isEditing.value = true),
```

File: packages/kotti-ui/source/kotti-comment/components/CommentReply.vue [hunk 5]
```diff
@@ -1,131 +1,103 @@
...
-}
+})
 </script>
+
 <style lang="scss" scoped>
-.comment-reply__message {
+.comment-reply {
 	display: flex;
-	align-items: center;
+	padding: 0.2rem 0;
```
Comment (santiagoballadares) [inline packages/kotti-ui/source/kotti-comment/components/CommentReply.vue:86]:
```suggestion
	padding: var(--unit-1) 0;
```

File: packages/kotti-ui/source/kotti-comment/components/CommentReply.vue [hunk 6]
```diff
@@ -1,131 +1,103 @@
...
+
+	&__content {
+		display: flex;
+		flex-direction: column;
+		width: 100%;
+		margin-left: 0.4rem;
```
Comment (santiagoballadares) [inline packages/kotti-ui/source/kotti-comment/components/CommentReply.vue:92]:
```suggestion
		margin-left: var(--unit-2);
```

File: packages/kotti-ui/source/kotti-comment/components/CommentReply.vue [hunk 7]
```diff
@@ -1,131 +1,103 @@
...
+
+		&__info {
+			display: flex;
+			flex-direction: row;
+			margin-bottom: 0.1rem;
+			font-size: 0.7rem;
+			line-height: 0.7rem;
```
Comment (santiagoballadares) [inline packages/kotti-ui/source/kotti-comment/components/CommentReply.vue:99]:
```suggestion
			margin-bottom: var(--unit-h);
			font-size: calc(var(--unit-3) + var(--unit-h));
			line-height: calc(var(--unit-3) + var(--unit-h));
```

File: packages/kotti-ui/source/kotti-comment/components/CommentReply.vue (hunk 2 above)
Comment (carsoli) [reply packages/kotti-ui/source/kotti-comment/components/CommentReply.vue:65]: great catch!

File: packages/kotti-ui/source/kotti-comment/KtComment.vue (hunk 1 above)
Comment (carsoli) [reply packages/kotti-ui/source/kotti-comment/KtComment.vue:161]: for font-sizes, i don't think we use unit- , i think we use those for spacings (margins , paddings) only

there were variables for the font but they are not prefixed with font sadly

`--font-size` , `font-size-sm`, `font-size-lg`

Comment (carsoli) [reply packages/kotti-ui/source/kotti-comment/KtComment.vue]: the line-height (I've never also seen it used with units) but we can make a variable for our root-line-height (one may already exist)

**Pull request merged**